packages/overlay/src/integrations/sentry/index.ts

```diff
@@ -143,16 +143,66 @@ export function processEnvelope(rawEvent: RawEventContext) {
   };
 }
 
+/**
+ * Takes care of injecting spotlight-specific behavior into the Sentry SDK by
+ * accessing the global __SENTRY__ carrier object.
+ *
+ * This is admittedly extremely hacky but it's the only way to hook into the SDK
+ * without requiring users to manually register a spotlight integration or unnecessarily
+ * increasing production build bundle size.
+ *
+ * Specifically, we:
+ * - Enable SDK integrations if the SDK is configured without a DSN. This way, we
+ *   can still capture events _only_ for spotlight
+ * - Add a spotlight Sentry integration to the SDK that forwards events to the
+ *   the spotlight sidecar ({@link spotlightIntegration})
+ *
+ * @param options options of the Sentry integration for Spotlight
+ */
 function addSpotlightIntegrationToSentry(options?: SentryIntegrationOptions) {
-  if (options?.injectIntoSDK === false) return;
-  // A very hacky way to hook into Sentry's SDK
-  // but we love hacks
-  const sentryHub = (window as WindowWithSentry).__SENTRY__?.hub;
-  const sentryClient = sentryHub?.getClient();
-  if (sentryClient) {
-    const spotlightIntegration = new Spotlight({
+  if (options?.injectIntoSDK === false) {
+    return;
+  }
+
+  const sentryGlobal =
+    // This is what we expect the v8-stable accessor to be
+    (window as WindowWithSentry).__SENTRY__?.acs?.getCurrentScope() ||
+    // This is the current accessor (v7 and v8-alpha)
+    (window as WindowWithSentry).__SENTRY__?.hub;
+
+  if (!sentryGlobal) {
+    warn("Couldn't find the Sentry SDK on this page. Make sure you're using a Sentry SDK with version 8.0.0 or higher");
+    return;
+  }
+
+  const sentryClient = sentryGlobal.getClient();
+  if (!sentryClient) {
+    warn("Couldn't find a Sentry SDK client. Make sure you're using a Sentry SDK with version 8.0.0 or higher");
+    return;
+  }
+
+  if (!sentryClient.getOptions().dsn) {
```

Comment:
Hi @Lms24 we should not force the user to remove DSN altogether to use spotlight.

Reply:
That's not what we're doing here. We only check if users configured Sentry without a DSN. If that's the case, the SDK will not initialize integrations by default (e.g. the ones that instrument the global error handlers). So in order for Spotlight to pick up events from the host app, we need to initialize the integrations "manually". 
If users already set a DSN, there's nothing for us to do other than adding the spotlight integration (see code below).

Reply:
Oh, i got it. When dsn is set, sdk will initialize the integrations. 

But what about invalid DSN?

Actually, i tested this locally with dsn value pointed to env variable, and with env variable value empty
And for me, integrations were not intialized.

Reply:
Let me check if integrations were not intialized or spotlight integration was not added. 
Because spotlight was not retrieving any envelopes.

Reply:
Yes @Lms24 i was adding an invalid DSN
So let's use `getDsn` instead of `getOptions.dsn` because it looks like we validate `dsn` in SDK before setting `_dsn` in senryClient.

```suggestion
  if (!sentryClient.getDsn()) {
```

Reply:
good call! I was experimenting with emulating the `isEnabled` check but it's probably better to limit the check to the DSN since users might want to explicilty disable the SDK via `enabled: false`.

Reply:
Updated in [c43f007](https://github.com/getsentry/spotlight/pull/386/commits/c43f0074482b2b70e8493124133596d79d0f1f57)